rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1125,7 +1125,7 @@ stream {
             {{ buildInfluxDB $location.InfluxDB }}
 
             {{ if not (empty $location.Redirect.URL) }}
-            if ($uri ~* {{ $path }}) {
+            if ($uri {{ if not $usesRegex }} ~* {{end}} {{ $path }}) {
```

Comment:
so when usesRegex is true, it will look like
```
if ($uri {{ $path }}) {
```
? What does this mean?

It'd be great to use a template function for this as well. It's hard to parse template, and doing logic here is error prone.

Reply:
`$uri ~* {{ $path }}` determines if `$uri` matches the regular expression string `$path`, where `$path` is the string returned from `buildLocation` within `internal/ingress/controller/template/template.go`. If it's a match, it returns true, otherwise it returns false. 

This is problematic when `buildLocation` returns a path with the location modifier because the location modifier is also `~*`, so `$uri ~* {{ $path }}` resolves into `$uri ~* ~* ^/pathname`.

I see what you mean with this being flakey. Instead, we could have a function in `internal/ingress/controller/template/template.go` that returns the path with the location modifier removed if it exists. 

More info: https://www.digitalocean.com/community/questions/conditional-within-an-if-in-nginx